src/plugins/core/formatters/html/enforce-p-elements.js

```diff
@@ -21,69 +19,41 @@ define([
 
   'use strict';
 
-  /**
-   * Wrap consecutive inline elements and text nodes in a P element.
-   */
-  function wrapChildNodes(scribe, parentNode) {
-    var groups = Array.prototype.reduce.call(parentNode.childNodes,
-                                             function (accumulator, binChildNode) {
-      var group = last(accumulator);
-      if (! group) {
-        startNewGroup();
-      } else {
-        var isBlockGroup = scribe.element.isBlockElement(group[0]);
-        if (isBlockGroup === scribe.element.isBlockElement(binChildNode)) {
-          group.push(binChildNode);
-        } else {
-          startNewGroup();
-        }
-      }
-
-      return accumulator;
-
-      function startNewGroup() {
-        var newGroup = [binChildNode];
-        accumulator.push(newGroup);
+  return function () {
+    return function (scribe) {
+      var nodeHelpers = scribe.node;
+      var elementHelpers = scribe.element;
+
+      /**
+       * Wrap consecutive inline elements and text nodes in a P element.
+       */
+      function wrapChildNodes(parentNode) {
+        var index = 0;
+        Immutable.List(parentNode.childNodes)
+          .filter(function(node) {
+            return node.nodeType === Node.TEXT_NODE || !elementHelpers.isBlockElement(node);
+          })
+          .groupBy(function(node, key, list) {
+            return key === 0 || node.previousSibling === list.get(key - 1) ?
+              index :
+              ++index;
```

Comment:
Can you use postfix notation here, prefix will not make any difference here speedwise as the compile will just optimise it away?